### ISSUE TITLE
Disabling `noImplicitAny` suppresses index type checking in bracket notation

### ISSUE THREAD

**Issue by pushkine** [microsoft/TypeScript#43323]
TS 4.2.3 and nightly
```ts
const obj = {};
obj.foo = 1;
obj["foo"] = 1;
```
* ["noImplicitAny": true](https://www.typescriptlang.org/play?ts=4.2.3#code/MYewdgzgLgBCBGArGBeGBvAvgbgLACgFEA6AMxBFRgEY9CkBtAInJCYF0rag)
```
[2] Property 'foo' does not exist on type '{}'.(2339)
[3] Element implicitly has an 'any' type because expression of type '"foo"' can't be used to index type '{}'
[3] Property 'foo' does not exist on type '{}'.(7053)
```
* ["noImplicitAny": false](https://www.typescriptlang.org/play?noImplicitAny=false&ts=4.2.3#code/MYewdgzgLgBCBGArGBeGBvAvgbgLACgFEA6AMxBFRgEY9CkBtAInJCYF0rag)
```
[2] Property 'foo' does not exist on type '{}'.(2339)
```
Expected an error in both cases regardless of `noImplicitAny`

**Comment by RyanCavanaugh** [comment]
To expedite the triage process, we need everyone to follow the issue template and instructions.

When you clicked "Create New Issue", the issue form was pre-populated with a template and some instructions. We need you to read those instructions *completely*, follow them, and then fill in *all* the fields in that template.

We are not able to assist with issues that don't follow the template instructions as they represent a significantly larger amount of work compared to issues which are correctly specified. Thank you for understanding.
